### PR TITLE
Checklist: Add new AMP validation check

### LINKED ISSUE #8442: Add AMP Error check to PPC
Adds a new check to the checklist for stories that are published that will show a checklist priority card if there are AMP errors not covered by other checks present in the story. 

Includes a link in copy to go test the story URL to get more details externally. 

See epic for more info.

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Based on #6184 to show when there are AMP violations in a story that has been published. 

## Summary

Adds a new check for if there are AMP errors present in a story. There must be either a story link present or the status is published/scheduled. 

Screen Shot 2021-07-28 at 2 48 36 PM

## Relevant Technical Choices

- Pascal did most of this in his proof of concept (#6184), I implemented in the new checklist structure. 
- Amp Error Validation is a new **priority** check. It updates on initial render and then whenever the `isSaving` prop from the `story` becomes `false` again. Right now, the actual check for errors will only run with a story has been published or scheduled. This is an intentional choice because 'saving' fixes most if not all errors with the AMP code. 
- We are filtering out errors in the validator that are checked internally. These are: missing story poster, missing publisher logo, missing video posters. The other priority check that is covered here is the story title, but AMP allows an empty title, so our check is better and that error never sees the light of day for the validator (as far as I can tell) 
- I have put this check behind its own feature flag because I'm not sure all of the issues we're filtering for are covered yet and it feels like something we might want to have the ability to exclude. Experiment is called `enableAmpValidationCheck`, set it to default true.

Fix three small bugs in checklist while making sense of the logic here: 
- the `storyPageCount` was digging one level too deep for `pages` and never rendering - pretty sure this was just a sneaky rebase
- publish/update button was keeping the exclamation icon and tooltip even when all checkpoints were hit and issues resolved. Limiting based on if the dialog should appear when checkpoint ALL is hit (also updating the `null` to an empty string for tooltip to silence a console warning about the propType. 
- update Panel sections to only set `isExpanded` to true when there's an accompanying issue count and the `isOpen` boolean is true, this was causing a little hiccup where if a section was expanded when all issues were solved and now the checklist is empty the helper text associated with that panel wouldn't be hidden. 
Screen Shot 2021-07-28 at 11 23 53 AM

## To-do

- [x] Need to add some tests for the check
- Look into Web Assembly implementation (new ticket here: https://github.com/google/web-stories-wp/issues/8477)
- [x] Finalize copy
- [ ] poster images for videos should trigger an error but can't find it! 

## User-facing changes

If you have AMP errors in your published/scheduled story you see this: 

Screen Shot 2021-07-29 at 12 06 23 PM

## Testing Instructions

Have a bad story and see the error. This is surprisingly hard to trigger. 
this is a list of AMP errors: https://amp.dev/documentation/guides-and-tutorials/learn/validation-workflow/validation_errors/?format=websites#amp-html-tag-and-attribute-errors  if you find a way to get this to fire please share! I was having luck before moving the check to be post publish. All errors I've found get resolved on save. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

- I added a new `trackClick` (`click_checklist_amp_test`) for the link that sends users to `https://search.google.com/test/amp` to test their story and get more robust details about what went wrong in their story. I figure this is useful to see if people are following through on testing stories with issues or if we need a more involved solution for walking users through fixing these errors. 
- Also added a new `trackEvent` (`amp_validation_errors_found`) which fires once per session when amp validation errors are found. 

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8442

### CODE REVIEW

File: assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          return false;
+        }
+        // AMP attribute is removed by the plugin if there is no cover image.
+        if (
+          'MANDATORY_ATTR_MISSING' === code &&
+          params[0] === '⚡' &&
+          params[1] === 'html'
+        ) {
+          return false;
+        }
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: We no longer do this, so this could be removed

File: assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { ChecklistCard, DefaultFooterText } from '../../checklistCard';
+import { PRIORITY_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function storyHasAmpErrors({ link, status }) {
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: Maybe `getAmpValidationErrors`?

It's otherwise a bit confusing because the React component has the same name, just with PascalCase.

File: assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        return true;
+      });
+    return filteredErrors.length > 0;
+  } catch {
+    return false;
+  }
+}
+
+const StoryHasAmpErrors = () => {
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: Besides the `trackClick`, I think we'd want to `trackEvent()` when the story has errors.

Perhaps that can be done in a `useEffect(() => if (isRendered){...})`? Ideally only once per session as we don't want to trigger that every time the user saves.

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: yeah totally can add that, good idea.

File: assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: totally get that, this is a pattern with all the other checks too - i suppose the thinking here was that both the function and the React component create the total `check`.   Open to changing it, especially if it's confusing to someone who hasn't been in that section of the code base as much as pea pod has.

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyHasAmpErrors.js]: 🤝  went with `getStoryAmpValidationErrors` and `StoryAmpValidationErrors`

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { ChecklistCard, DefaultFooterText } from '../../checklistCard';
+import { PRIORITY_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function getStoryAmpValidationErrors({ link, status }) {
+  // TODO: Enable for drafts as well?
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: I am not familiar enough to know if we should enable this for drafts as well. To me, seems like something that could go either way.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    const storyMarkup = await response.text();
+    const { status: markupStatus, errors } =
+      window.amp.validator.validateString(storyMarkup);
+
+    if ('FAIL' !== markupStatus) {
+      return false;
+    }
+
+    const filteredErrors = errors
+      .filter(({ severity }) => severity === 'ERROR')
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:46]: these strings are only used here in this filtering, didn't feel necessary to pull them out as constants - lemme know if you disagree and i'll update.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      .filter(({ code, params }) => {
+        // Filter out errors that are covered in other checks
+        // Already covered by metadata checks.
+        if ('MISSING_URL' === code && params[0].startsWith('poster')) {
+          return false;
+        }
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: We probably need to add more here, for example to filter out the publisher logo warnings.

File: assets/src/edit-story/components/checklist/constants.js [hunk 7]
```diff
@@ -469,6 +475,33 @@ export const PRIORITY_COPY = {
...
+          ),
+        }}
+      >
+        {__(
+          'Your story contains errors that make it ineligible to appear in dedicated places on Google Search and Discover. For more insight on how to resolve these errors, visit <a>Google AMP Test</a>.',
+          'web-stories'
+        )}
+      </TranslateWithMarkup>
+    ),
+  },
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/constants.js]: @o-fernandez @choumx Thoughts on wording here and where we should point users to?

Comment (o-fernandez) [reply assets/src/edit-story/components/checklist/constants.js]: My suggestion (minor edits to make it shorter and add a "may" in case some errors don't prevent distro): 

"Your story contains errors that may reduce where it is seen on Google Search and Discover. To learn more, visit <a>Google AMP Test</a>"

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: I'll look closer at the AMP documentation today and try to filter out some more. Asked @choumx to take a look as he's familiar with AMP.

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: thanks! will update

Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/constants.js]: Per [docs](https://developers.google.com/search/docs/advanced/appearance/enable-web-stories#google-discover), there may be additional surfaces:
> Web Stories can appear on Google across Search, Google Discover, and Google Images...

How about "can affect eligibility to appear on Google"? I.e.

"Your story contains errors that can affect eligibility to appear on Google. To learn more, visit Google AMP Test"

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: I think filtering on a validation error-by-error basis will be difficult and probably brittle.

Could we make triggering of the entire check conditional instead? If we only trigger the AMP validation PPC when there are no high priority PPC items, then we know that any validation errors found are not duplicative.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 4 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: I think enabling for drafts is probably less confusing. Do we have any other PPC items that don't run on drafts?

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: nope! everything runs on drafts otherwise. i thought maybe this rule here was mentioned because the validation check doesn't work well on a draft since those aren't done? 

we can swap it. have anything with a link run.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: 🤔  

so, that would mean the following set of actions/responses
- story reaches checkpoint so priority are visible (publish attempt or more than 4 pages)
- I fix all my priority issues 
- amp validation error pops up *is this a priority issue or a new section*?
- If a high priority item returns (i don't know know why you'd do this but let's say you remove your title), does the amp validation error message disappear?

Comment (swissspidy) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: There are only 4 priority checks related to issues that would cause AMP invalidation.

* Missing title
* Missing story poster
* Missing publisher logo
* Missing video posters

Filtering by that seems pretty straightforward. I can do a quick pass if you'd like.

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: This look ok? 

Screen Shot 2021-07-27 at 2 14 50 PM

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: so that'll work if we can move this check to be anywhere besides the priority panel. It gets goofy with the logic if we are trying to render something in the priority panel that isn't counted as a priority issue and should only show up when the priority panel has been resolved and otherwise wouldn't be open.

Thoughts on it existing in the accessibility panel? 
Screen Shot 2021-07-27 at 2 27 29 PM

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (swissspidy) [reply assets/src/edit-story/components/checklist/constants.js]: Let‘s change link to „
Web Stories Test Tool“

So „… visit the 
Web Stories Test Tool“

The docs refer to it as the 
Web Stories Google Test Tool, but that‘s a bit long

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (swissspidy) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: That sounds like an odd category for this check. Perhaps a new one is needed? e.g. „Distribution“ or „Validation“

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: So testing this a bit more and I'm a bit confused. 

If you take a template and make a story out of it and clear up the priority issues, you'll see the new amp error check show up  and it has the following codes causing it to show up: `MISSING_REQUIRED_EXTENSION`, `DISALLOWED_SCRIPT_TAG`, `WRONG_PARENT_TAG`, `MANDATORY_TAG_MISSING` but they are all resolved the moment you save the story or publish it.


This leads me to think that we should only check published stories, not drafts as it's not a relevant issue to show to people until they have published. Alternately, could track for first save and start showing it then, but it feels like a check that should happen once the story is done.

https://user-images.githubusercontent.com/10720454/127231857-0797855c-6b9c-4236-befa-9e6c52c8123b.mp4



Thoughts @choumx @swissspidy ?

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: Screen Shot 2021-07-27 at 2 49 48 PM

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: > That sounds like an odd category for this check. Perhaps a new one is needed? e.g. „Distribution“ or „Validation“

Agree, @swissspidy - let's do distribution

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: So the video above is still applicable, just pretend it shows up in this new panel. I think this is something that only needs to show up on publish because most of these errors are resolving on save, so they'd only be relevant if they stick around after publishing. 

Am i on the right track? 

Screen Shot 2021-07-27 at 3 01 18 PM

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: also, @choumx or @o-fernandez please someone make that copy better than `Make your Web Story distributable`

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 4 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: (see my comment on the below thread, i think this should stay at just after publish. Most if not all issues I've been able to get in the editor resolve on save or publish.)

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: > There are only 4 priority checks related to issues that would cause AMP invalidation.... Filtering by that seems pretty straightforward. 

Hmm, we can still try filtering errors if the performance of validation is acceptable. A nice thing about limiting triggering (e.g. 0 priority items and on-publish) is that it helps constrain performance concerns.

Regardless, we should highlight this check so that users are aware of issues before publishing. That means making sure it's treated the same as high priority items in the blocking popup when clicking publish, etc.

(I can even imagine an alternative UX where this check **only** lives in that blocking popup instead of as a new PPC item -- would that be simpler?) 

> If a high priority item returns (i don't know know why you'd do this but let's say you remove your title), does the amp validation error message disappear?

I agree this seems a bit weird but maybe it's OK since this AMP validation check should just be a rare catch-all for configuration/environment issues like conflicting plugins, etc.

This also should inform the copy we use -- IMO we should:
- Suggest possible issues with the creator's WP setup (in the check or in the FAQ/help article)
- Link to the FAQ or forums instead of the AMP Test Tool (which was built for developers)
- Consider avoiding developer terms like "validation" ("compatibility" instead?)

> Fix compatibility errors to make your story eligible to appear on Google.
> ### Compatibility
> Your story contains errors that can affect eligibility to appear on Google. This can be caused by incompatible WordPress plugins, web application firewalls, or other hosting configuration issues. Please see [our FAQ](https://wordpress.org/plugins/web-stories/#what%20do%20i%20do%20about%20an%20amp%20validation%20issue%20in%20search%20console%3F) for more information and debugging tips.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+import { DISTRIBUTION_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function getStoryAmpValidationErrors({ link, status }) {
+  if (!link || !['publish', 'future'].includes(status)) {
+    return false;
+  }
+
+  try {
+    const response = await fetch(link);
```
Comment (dreamofabear) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:36]: Does this fetch the story markup that would ultimately be rendered? I.e. including all WP filters in PHP, etc.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: > This leads me to think that we should only check published stories, not drafts as it's not a relevant issue to show to people until they have published.

That sounds fine to me too.

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/constants.js]: On second thought, the Test Tool is quite technical so perhaps we shouldn't link to it. Continued here: https://github.com/google/web-stories-wp/pull/8443#discussion_r677870166

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: > Hmm, we can still try filtering errors if the performance of validation is acceptable. A nice thing about limiting triggering (e.g. 0 priority items and on-publish) is that it helps constrain performance concerns. 

the way I've got it now is the validation check will only run once there are no priority issues left, additionally it's only ever updating when `isSaved` is triggered so i'd say performance isn't something we need to be really worried about on this one here. 

> we should highlight this check so that users are aware of issues before publishing. That means making sure it's treated the same as high priority items in the blocking popup when clicking publish, etc. 

This is going to be more involved than just adding a new check. 

It _feels like_ this is a priority check. I'm hesitant to stick with a new panel. If it's really only 4 priority checks that we need to check and see if they exist before rendering the AMP check, then I think just checking internally how we had it earlier would be the way to go. Since the check's memoized and only running on `isSaved` it's not contingent on the entire story updating all the time. It wouldn't be a big deal, perf wouldn't be impacted. That would make sure users have the same shot at reviewing it as other priority issues. 

Thoughts? Given our approaching release we should either move this to the next release if we want its own panel with new dialog logic and triggers or we should put it in the priority panel and limit the check internally which i can wrap up tomorrow.

Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: If you want to jump on a call in the AM we can work it out that way too.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 8 above)
Comment (swissspidy) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:36]: Yes

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: Screen Shot 2021-07-28 at 2 48 36 PM
Screen Shot 2021-07-28 at 2 50 35 PM


@choumx thoughts?

File: assets/src/edit-story/components/checklist/constants.js [hunk 9]
```diff
@@ -469,6 +475,33 @@ export const PRIORITY_COPY = {
...
     ),
   },
+  ampValidation: {
+    title: __('Compatibility', 'web-stories'),
+    footer: (
+      <TranslateWithMarkup
+        mapping={{
+          a: (
+            <Link
+              href="https://wordpress.org/plugins/web-stories/#what%20do%20i%20do%20about%20an%20amp%20validation%20issue%20in%20search%20console%3F"
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/constants.js]: as far as I can tell, this isn't translatable.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: ok dokey! 

@choumx @swissspidy  I updated. Can you take a peak?  I updated the PR description to be central to this thread and the decisions made. It ended up being just barely different from where we started yesterday AM 😅. 

### TL;DR
- AMP Validation Error check is part of priority group
- Filtering out for known priority issues (publisher logo, story poster, video poster, title (this one happens naturally it seems? I couldn't get it to fire, if you can awesome.)) 
- Check will only ever update 1) initial render 2) whenever `isSaving` becomes false again, this means that any issues that could have been present but would be cleaned up on a save or publish are now gone so it silences some unnecessary noise. 
 _A note on the above_: This is something you can see happen if you create a story from a template and unblock the validation check for drafts/save, but when you publish the template story then they all go away. 
- since it's included in priority check, if a story is scheduled or published and there's some errors hanging on then the little exclamation icon will stay present with the tooltip to remind user to check it out (or if they haven't seen the checklist at all, that dialog will show). 

Anyways, this is now behind a feature flag (defaulted to true) and we've got some event tracking set up. I'm curious how often this even fires? Feels like a good first implementation. 

Let me know what you think and [if the wording](https://github.com/google/web-stories-wp/pull/8443#discussion_r678676306) is appropriate or if there's other specific filters we should add to the check that I am unaware of. 

Thanks!

File: assets/src/edit-story/components/checklist/constants.js (hunk 9 above)
Comment (swissspidy) [reply assets/src/edit-story/components/checklist/constants.js]: It is. See https://br.wordpress.org/plugins/web-stories/ for example.

But I don't think we should link to this page here, honestly.

We should link to https://wp.stories.google/docs/ instead where have the same content.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+        // Missing publisher logo
+        if (
+          'MISSING_URL' === code &&
+          params?.[0].startsWith('publisher-logo')
+        ) {
+          return false;
+        }
+        // Missing video posters
+        if ('INVALID_URL_PROTOCOL' === code && params?.[0].startsWith('src')) {
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:64]: Hmm video posters are set as `<amp-video poster="...">`, not in a `src` attribute.

Need to check for the `poster` attribute and that it's an `amp-video` node.

File: assets/src/edit-story/components/checklist/constants.js [hunk 11]
```diff
@@ -469,6 +475,33 @@ export const PRIORITY_COPY = {
...
+              rel="noreferrer"
+              target="_blank"
+              aria-label={__(
+                'Web Stories frequently asked questions',
+                'web-stories'
+              )}
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/constants.js]: Don't think we need an aria-label here? The link text is pretty clear

File: assets/src/edit-story/components/checklist/constants.js [hunk 12]
```diff
@@ -469,6 +475,33 @@ export const PRIORITY_COPY = {
...
+              onClick={onAmpErrorClick}
+              size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}
+            />
+          ),
+        }}
+      >
+        {__(
+          'Your story contains errors that can affect eligibility to appear on Google. This can be caused by incompatible WordPress plugins, web application firewalls, or other hosting configuration issues. Please see <a>our FAQ</a> for more information and debugging tips.',
```
Comment (swissspidy) [inline assets/src/edit-story/components/checklist/constants.js]: This sentence is a) quite long and b) contains more info than the FAQ entry 😅 

Shouldn't we just link to the forums instead in that case?

File: includes/Experiments.php [hunk 13]
```diff
@@ -231,6 +231,18 @@ public function get_experiments(): array {
...
 			],
+			/**
+			 * Author: @brittanyirl
+			 * Issue: 8443
+			 * Creation date: 2021-07-28
+			 */
+			[
+				'name'        => 'enableAmpValidationCheck',
+				'label'       => __( 'Amp Validation Error Check', 'web-stories' ),
+				'description' => __( 'Enable AMP validation error in the checklist', 'web-stories' ),
```
Comment (swissspidy) [inline includes/Experiments.php]:
```suggestion
				'description' => __( 'Enable AMP validation in the checklist', 'web-stories' ),
```

File: assets/src/edit-story/components/checklist/constants.js (hunk 9 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: Thank you! This project sometimes makes me acutely aware of my "American" perspective and ignorances :) 

I'll update now.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 10 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:64]: i thought this might be wrong. So when I take the poster out of a video this is the error I'm getting, nothing else. I'll recreated and include some data.

File: assets/src/edit-story/components/checklist/constants.js (hunk 7 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/constants.js]: Looks good to me! /cc @o-fernandez

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 10 above)
Comment (swissspidy) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:64]: Interesting 🤔

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 6 above)
Comment (dreamofabear) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: > title (this one happens naturally it seems? I couldn't get it to fire, if you can awesome.

I think AMP validation checks for existence of the `amp-story[title]` attribute while our PPC check looks for empty string: https://github.com/google/web-stories-wp/blob/d093d9f0980d4a3b2a8c5ed2a669308a95993eb9/assets/src/edit-story/output/story.js#L74-L80

No changes needed I think.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+  }
+}
+
+const StoryAmpValidationErrors = () => {
+  // ampValidationErrorRef is making sure that tracking is only fired once per session.
+  const ampValidationErrorsRef = useRef();
+  const {
+    meta: { isSaving },
+    story: { link, status },
+  } = useStory(({ state }) => state);
```
Comment (dreamofabear) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js]: Should we use a more specific selector for `useStory` (only link/status/isSaving)? I think this check will re-render on every selection change as is.

File: assets/src/edit-story/components/checklist/constants.js (hunk 12 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/constants.js]: 😂 

Shortened to this? 
> Your story contains errors that can affect eligibility to appear on Google. Please see our FAQ for more information.

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js (hunk 10 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:64]: ok so i've deleted the featured image from the video in media and i've gutted the poster url and id in the editor dev tools.

```
resource": {
            "type": "video",
            "mimeType": "video/mp4",
            "creationDate": "2021-07-28T17:41:49",
            "src": "http://localhost:8899/wp-content/uploads/2021/07/mitch.mp4",
            "width": 610,
            "height": 514,
            "poster": "",
            "posterId": null,
            "id": 920,
            "length": 1,
            "lengthFormatted": "0:01",
            "title": "mitch",
            "alt": "mitch",
            "sizes": [],
            "local": false,
            "isPlaceholder": false,
            "isOptimized": false,
            "isMuted": false,
            "provider": "local",
            "baseColor": [
              76,
              77,
              97
            ]
          },
```

this is the only error I am getting before filtering out anything:
Screen Shot 2021-07-29 at 11 19 38 AM

File: assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+  }));
+
+  // isRendered is getting set asynchronously based on the returned value of `getStoryAmpValidationErrors`,
+  // setting this statefully allows the value to be a boolean not a promise which is crucial to updating tracking accurately.
+  const [isRendered, setIsRendered] = useState(false);
+
+  // using `isSaving` as an indicator of when to check for validation errors so that this check doesn't run on all story updates.
+  // Allows us to just track status and link, puts the story in line with what the outcome would be if the user was testing their
+  // story link in the recommended test/amp site.
+  useEffect(() => {
```
Comment (maxyinger) [inline assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js:92]: excellent comments! 👏

**Review by maxyinger — APPROVED** [review]
Looks good to me. Couldn't figure out how to trigger it beyond commenting out the check for  publisher logo error.

But everything here looks good! Thanks for all the smoll bug fixes you included and great comments

**Review by samwhale — APPROVED** [review]
This is great, and excellent comments 👏

**Comment by swissspidy** [comment]
@BrittanyIRL FYI there are now some merge conflicts

File: includes/Experiments.php [hunk 16]
```diff
@@ -219,6 +219,18 @@ public function get_experiment_groups(): array {
...
+			 * Author: @brittanyirl
+			 * Issue: 8443
+			 * Creation date: 2021-07-28
+			 */
+			[
+				'name'        => 'enableAmpValidationCheck',
+				'label'       => __( 'Amp Validation Error Check', 'web-stories' ),
+				'description' => __( 'Enable AMP validation in the checklist', 'web-stories' ),
+				'group'       => 'editor',
+				'default'     => true,
```
Comment (swissspidy) [inline includes/Experiments.php]: What's the point of adding this as an experiment if it's enabled by default from the beginning? 🤔

Comment (BrittanyIRL) [reply includes/Experiments.php]: my thinking was that since this is hard to trigger, it might be nice to have an off switch if something unexpected happened. probably redundant but thought it wouldn't hurt.

Comment (BrittanyIRL) [reply includes/Experiments.php]: meh, there's no good reason - i'm scrapping it.
